react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,44 @@
+import { NextButton } from './NextButton';
+import { PrevButton } from './PrevButton';
+import { Dots } from './Dots';
+import { ImageCard } from './ImageCard';
+import { images } from './App';
+import { useEffect, useState } from 'react';
+type images = {
+  src: string;
+  alt: string;
+};
+type Props = {
+  images: images[];
+};
+
+export function Carousel({ images }: Props) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setCurrentIndex((currentIndex) => (currentIndex + 1) % images.length);
+    }, 2000);
+    return () => clearTimeout(timerId);
+  }, [currentIndex, images.length]);
```

Comment:
`currentIndex` isn't a dependency in this function. On line 20, in your `setCurrentIndex` call, you are passing it an arrow function. That arrow function has one parameter, which you've named `currentIndex`. So the `currentIndex` variable on line 16 is not actually being used inside the effect function.

This is one reason I like to use `prev` when I use that form of the setter. I think it's less likely to confuse me!:
`setCurrentIndex((prev) => (prev + 1) % images.length);`